Examples/Tests/MacroExamples/Implementation/Member/MetaEnumMacroTests.swift

```diff
@@ -0,0 +1,120 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import MacroExamplesImplementation
+import SwiftSyntaxMacros
+import SwiftSyntaxMacrosTestSupport
+import XCTest
+
+final class MetaEnumMacroTests: XCTestCase {
+  private let macros = ["MetaEnum": MetaEnumMacro.self]
+
+  func testExpansionAddsNestedMetaEnum() {
+    assertMacroExpansion(
+      """
+      @MetaEnum enum Cell {
+        case integer(Int)
+        case text(String)
+        case boolean(Bool)
+        case null
+      }
+      """,
+      expandedSource: """
+        enum Cell {
+          case integer(Int)
+          case text(String)
+          case boolean(Bool)
+          case null
+
+            enum Meta {
+                    case integer
+                    case text
+                    case boolean
+                    case null
+
+                    init(_ __macro_local_6parentfMu_: Cell) {
+                        switch __macro_local_6parentfMu_ {
+                        case .integer:
+                            self = .integer
+                        case .text:
+                            self = .text
+                        case .boolean:
+                            self = .boolean
+                        case .null:
+                            self = .null
+                        }
+                    }
```

Comment:
You should be able to fix this formatting by changing MetaEnumMacro to

```swift
  func makeMetaEnum() -> DeclSyntax {
    // FIXME: Why does this need to be a string to make trailing trivia work properly?
    let caseDecls = childCases.map { childCase in
      "  case \(childCase.name)"
    }.joined(separator: "\n")

    return """
      \(access)enum Meta {
      \(raw: caseDecls)
      \(makeMetaInit())
      }
      """
  }

  func makeMetaInit() -> DeclSyntax {
    // FIXME: Why does this need to be a string to make trailing trivia work properly?
    let caseStatements = childCases.map { childCase in
      """
        case .\(childCase.name):
          self = .\(childCase.name)
      """
    }.joined(separator: "\n")

    return """
      \(access)init(_ \(parentParamName): \(parentTypeName)) {
        switch \(parentParamName) {
      \(raw: caseStatements)
        }
      }
      """
  }
```

Reply:
Done in https://github.com/apple/swift-syntax/pull/2163/commits/39032ab805de40a7a4c7416a71a3f7704662a8ce. Thanks!